### PR TITLE
Lower the embedding op

### PR DESCRIPTION
**Pull request by bhavya01**
Reference: https://github.com/pytorch/pytorch/blob/113138aa5575301d914c18bd882d6ab3735aa18a/aten/src/ATen/native/Embedding.cpp#L37

The native pytorch implmentation also uses just the indices and weight matrix. Other options seem to be ignored.

### CODE REVIEW

**Comment by wonjoolee95** [comment]
@bhavya01, is this ready to be reviewed? Also can you add a corresponding cpp unit test as it is a new op?

**Comment by bhavya01** [comment]
@wonjoolee95 This PR should be ready for a review for #5982

**Comment by wonjoolee95** [comment]
Do we already have a cpp test for this op? Also wil `embedding_bag` also be lowered in this PR or will that be in a follow-up PR?

**Comment by wonjoolee95** [comment]
@bhavya01, can we ensure that the lowering get invoked correctly by adding the metric check in the `embedding` cpp test like https://github.com/pytorch/xla/blob/master/test/cpp/test_aten_xla_tensor_5.cpp#L246? 

Thanks, let's also work on the `embedding_bag` as we have the customer request on that op. Thanks for working on this!

**Comment by bhavya01** [comment]
Added the check for metrics.  EmbeddingBag is still WIP.

File: test/cpp/test_aten_xla_tensor_5.cpp [hunk 1]
```diff
@@ -261,6 +261,9 @@ TEST_F(AtenXlaTensorTest, TestEmbedding) {
                                            /*scale_grad_by_freq=*/false,
                                            /*sparse=*/false);
     AllClose(b, xla_b);
+    ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
+    ExpectCounterChanged("xla::embedding_symint",
```
Comment (wonjoolee95) [inline test/cpp/test_aten_xla_tensor_5.cpp:265]: nit: `xla::embedding_syntint` -> `xla::embedding*`. Not blocker, we can update in the EmbeddingBag PR.

Comment (wonjoolee95) [reply test/cpp/test_aten_xla_tensor_5.cpp:265]: Also want to confirm one thing -- before implementing this lowering (i.e. without the changes in this PR), does this metric assertion fail as expected? Just wanted to make sure that the `_symint` variant is getting recognized properly. It should, but just wanted to confirm.

But not a blocker, we can follow-up in the EmbeddingBag PR.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 2]
```diff
@@ -3515,11 +3515,10 @@ at::Tensor XLANativeFunctions::embedding_symint(const at::Tensor& weight,
     return at::native::embedding_symint(weight, indices, padding_idx,
                                         scale_grad_by_freq, sparse);
   }
-  // TODO: for now route to native, which dispatches supported XLA operations.
-  // We need to make use of the TPU embedding core here eventually.
```
Comment (wonjoolee95) [inline torch_xla/csrc/aten_xla_type.cpp:3519]: This comment (seems like it existed for a while) `We need to make use of the TPU embedding core here eventually.` makes me a bit cautious about the `Embedding` op. I assume this comment means to lower this op in a performant way, we somehow want to make use of the TPU embedding core? Just leaving this comment as a future reference. Also let's bring this comment back in the code so we are aware. Again, not a blocker, we can include as a follow-up in the next PR.

**Review by wonjoolee95 — APPROVED** [review]
Thanks! 

I left a few comments as follow-up items. Not a blocker, so feel free to merge this PR. But let's follow-up on the items in the next PR (preferably the EmbeddingBag PR).

File: test/cpp/test_aten_xla_tensor_5.cpp (hunk 1 above)
Comment (bhavya01) [reply test/cpp/test_aten_xla_tensor_5.cpp:265]: Yes, without the change this check fails.

image

File: torch_xla/csrc/aten_xla_type.cpp (hunk 2 above)
Comment (bhavya01) [reply torch_xla/csrc/aten_xla_type.cpp:3519]: Sounds good. Will add it back in a follow up PR

**Pull request merged**